pipelines/zendesk/helpers/talk_api.py

```diff
@@ -0,0 +1,115 @@
+    "recordings": "/api/v2/channels/voice/calls/{call_id}/recordings",
+    "digital_lines": "/api/v2/channels/voice/digital_lines",
+    "agents_overview": "/api/v2/channels/voice/stats/agents_overview",
+    "account_overview": "/api/v2/channels/voice/stats/account_overview",
+}
+ZENDESK_STATUS_CODES = {
+    "ok": 200,
+    "rate_limit": 429
+}
+
+
+class ZendeskAPIClient:
+    """
+    API client used to make requests to Zendesk TALK.
+    """
+    subdomain: str = ""
+    url: str = ""
+    headers: DictStrStr = {}
+    auth: Tuple[str, TSecretValue]
+
+    def __init__(self, credentials: Union[ZendeskCredentialsOAuth, ZendeskCredentialsToken, ZendeskCredentialsEmailPass]) -> None:
+        """
+        Initializer for the API client which is then used to make API calls to the ZendeskAPI
+        @:param credentials: ZendeskCredentials object which contains the necessary credentials to authenticate to ZendeskAPI
+        @:returns: Nothing since it just initializes API parameters
+        """
+        # set subdomain, this is always needed to configure endpoints
+        self.subdomain = credentials.subdomain
+        self.url = f"https://{self.subdomain}.zendesk.com"
+
+        # oauth token is the preferred way to authenticate, followed by api token and then email + password combo
+        # if none of these are filled, simply return an error
+        if isinstance(credentials, ZendeskCredentialsOAuth):
+            self.headers["Authorization"] = f"Bearer {credentials.oauth_token}"
+        elif isinstance(credentials, ZendeskCredentialsToken):
+            self.headers["Authorizations"] = f"Bearer {credentials.token}"
+        elif isinstance(credentials, ZendeskCredentialsEmailPass):
+            self.auth = (credentials.email, credentials.password)
+
+    def make_request(self, endpoint: str, data_point_name: str) -> Iterator[TDataItems]:
+        """
+        Makes a get request on a given endpoint.
+        @:param endpoint: String that specifies the exact endpoint data is being received from
+        @:return response_json:
+        """
+        # TODO: 1 automatic retry on normal failures
+        # TODO: caching
+        # TODO: side loading
+        # TODO: handle incremental load
+        # make request and keep looping until there is no next page
+        get_url = f"{self.url}{endpoint}"
+        while get_url:
+            try:
+                response = requests.get(get_url, headers=self.headers, auth=self.auth)
+                if response.status_code == ZENDESK_STATUS_CODES["ok"]:
+                    # check if there is a next page and yield the response,
+                    # usually all relevant data is stored in a key with same name as endpoint
+                    response_json = response.json()
+                    get_url = response_json.get("next_page", None)
+                    yield response_json[data_point_name]
+                elif response.status_code == ZENDESK_STATUS_CODES["rate_limit"]:
```

Comment:
those are standard HTTP codes, you are OK with just using numbers

Reply:
I just left those for readability and making hackability of code easier for people who might not be as experienced with HTTP codes.